tests/datasets/test_metrics_processor.py

```diff
@@ -451,3 +468,46 @@ def test_metrics_polymorphic_processor(
             PolymorphicMetricsProcessor().process_message(message, meta)
             == expected_polymorphic_result
         )
+
+
+TEST_CASES_GENERIC = [
+    pytest.param(
+        SET_MESSAGE_SHARED,
+        [
+            {
+                "use_case_id": "release-health",
+                "org_id": 1,
+                "project_id": 2,
+                "metric_id": 1232341,
+                "timestamp": expected_timestamp,
+                "tags.key": [10, 20, 30],
+                "tags.indexed_value": [11, 22, 33],
+                "tags.raw_value": ["value-1", "value-2", "value-3"],
+                "metric_type": "set",
+                "set_values": [324234, 345345, 456456, 567567],
+                "materialization_version": 1,
+                "timeseries_id": ANY,
```

Comment:
I think we should test the time series id calculation works. 

Having the field populated here ensures at least that it is not None, but this is secondary problem.
I think you should have a dedicated unit test for the time series id generator that validates that messages with different tags keys/vals/metric id  do not share the same time series id.

Reply:
agreed, will add